dataactcore/read_config.py

```diff
@@ -0,0 +1,165 @@
+    CONFIG_BROKER['aws_role'] = None
+    CONFIG_BROKER['aws_create_temp_credentials'] = None
+    CONFIG_BROKER['aws_region'] = None
+
+    # if not using AWS and no error report path specified,
+    # default to `data_act_broker` in user's home dir
+    error_report_path = CONFIG_SERVICES['error_report_path']
+    if not error_report_path:
+        error_report_path = os.path.join(expanduser('~'), 'data_act_broker')
+    normpath(error_report_path)
+    CONFIG_SERVICES['error_report_path'] = error_report_path
+
+    # if not using AWS and no DynamoDB settings specified, default
+    # to localhost and port 8000
+    if not CONFIG_DB['dynamo_host']:
+        # TODO: dynamo host/port don't do anything; code assumes localhost/8000
+        CONFIG_DB['dynamo_host'] = '127.0.0.1'
+    if not CONFIG_DB['dynamo_port']:
+        CONFIG_DB['dynamo_port'] = 8000
+    # TODO: can we test that local dynamo is up and running? if not, route calls hang
+
+storage_path = CONFIG_BROKER['d_file_storage_path']
+if storage_path[-1] != os.path.sep:
+    CONFIG_BROKER['d_file_storage_path'] = "".join([storage_path, os.path.sep])
+
+# if no broker file path specified,
+# default to `data_act_broker` in user's home dir
+broker_files = CONFIG_BROKER['broker_files']
+if not broker_files:
+    broker_files = os.path.join(expanduser('~'), 'data_act_broker')
+elif len(os.path.splitext(broker_files)[1]):
+    # if config's broker_files is set to a actual filename
+    # just use the directory
+    broker_files = os.path.split(broker_files)[0]
+normpath(broker_files)
+if broker_files[-1] != os.path.sep:
+    broker_files += os.path.sep
+CONFIG_BROKER['broker_files'] = broker_files
+
+# normalize logging path, if given
+log_path = CONFIG_LOGGING['log_files']
+if log_path:
+    CONFIG_LOGGING['log_files'] = normpath(log_path)
+
+# we don't want http:// or ports in the host variables
+CONFIG_SERVICES['broker_api_host'] = re.sub(
+    'http://|:(.*)', '', CONFIG_SERVICES['broker_api_host'])
+CONFIG_SERVICES['validator_host'] = re.sub(
+    'http://|:(.*)', '', CONFIG_SERVICES['validator_host'])
+# if hosts in config file are set to 0.0.0.0, override to
+# 127.0.0.1 for cross-platform compatibility
+if CONFIG_SERVICES['broker_api_host'] == '0.0.0.0':
+    CONFIG_SERVICES['broker_api_host'] = '127.0.0.1'
+if CONFIG_SERVICES['validator_host'] == '0.0.0.0':
+    CONFIG_SERVICES['validator_host'] = '127.0.0.1'
+if CONFIG_DB['dynamo_host'] == '0.0.0.0':
+    CONFIG_DB['dynamo_host'] = '127.0.0.1'
+
+if CONFIG_SERVICES["broker_api_port"] == 443:
+    # Use https
+    CONFIG_SERVICES["protocol"] = "https"
+else:
+    CONFIG_SERVICES["protocol"] = "http"
+
+# Log some values from config
+for category_name in VALUES_TO_LOG:
+    category = globals()[category_name]
+    category_message = "### {}".format(category_name)
+    for key in VALUES_TO_LOG[category_name]:
+        value = category[key]
+        category_message = ", ".join([category_message, "{}:{}".format(key,value)])
```

Comment:
I'd recommend sticking to one form of string formatting here (_i.e._, either `join` or `format` but not both), perhaps something like `category_message = "{} {}: {}".format(category_message, key, value)`